include/supercluster.hpp

```diff
@@ -232,24 +252,31 @@ class Supercluster {
 
         Zoom() = default;
 
-        Zoom(const GeoJSONFeatures &features_) {
+        Zoom(const GeoJSONFeatures &features_, const Options &options_) {
             // generate a cluster object for each point
             std::uint32_t i = 0;
-
+            clusters.reserve(features_.size());
             for (const auto &f : features_) {
-                clusters.emplace_back(project(f.geometry.get<GeoJSONPoint>()), 1, i++);
+                auto cluster = Cluster(project(f.geometry.get<GeoJSONPoint>()), 1, i++);
+                const auto clusterProperties = options_.map(f.properties);
+                for (const auto &p : clusterProperties) {
+                    cluster.addProperty(p);
+                }
```

Comment:
Correspondingly, if we have a non-empty default `map`, should we avoid calling it (and adding any properties to the cluster) if `reduce` wasn't specified?

Reply:
That's true, I will update it.